### PR TITLE
Fixed #25175 -- Create an alias for the postgresql_psycopg2 database backend

### PR DESCRIPTION
**Pull request by caioariede**
https://code.djangoproject.com/ticket/25175

### CODE REVIEW

File: docs/internals/deprecation.txt [hunk 1]
```diff
@@ -399,6 +402,9 @@ details on these changes.
 * Private APIs ``django.db.models.sql.where.WhereNode.make_atom()`` and
   ``django.db.models.sql.where.Constraint`` will be removed.
 
+* ``django.db.backends.postgresql_psycopg2`` was renamed to
```
Comment (timgraham) [inline docs/internals/deprecation.txt]: Chop this note -- only removals are included here.

File: django/db/backends/postgresql_psycopg2.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import warnings
+
+from django.utils.deprecation import RemovedInDjango110Warning
+
+
+warnings.warn(
+    'The database backend for PostgreSQL was renamed from '
+    '\'django.db.backends.postgresql_psycopg2\' to '
+    '\'django.db.backends.postgresql\'.', RemovedInDjango110Warning)
```
Comment (timgraham) [inline django/db/backends/postgresql_psycopg2.py]: 110 -> 20

**Comment by timgraham** [comment]
It might be easier to allow 'postgresql_psycopg2' to continue to be used without an immediate deprecation. I think it will be a bit of a pain to configure Jenkins to vary the engine name based on the Django version.

File: docs/ref/settings.txt [hunk 3]
```diff
@@ -500,7 +500,7 @@ Default: ``''`` (Empty string)
 
 The database backend to use. The built-in database backends are:
 
-* ``'django.db.backends.postgresql_psycopg2'``
+* ``'django.db.backends.postgresql'``
```
Comment (timgraham) [inline docs/ref/settings.txt]: Add a note below this:

```
.. versionchanged:: 1.9

    The 'django.db.backends.postgresql' backend is named
    'django.db.backends.postgresql_psycopg2' in older releases.
```

File: docs/ref/settings.txt [hunk 4]
```diff
@@ -657,8 +657,7 @@ The character set encoding used to create the test database. The value of this
 string is passed directly through to the database, so its format is
 backend-specific.
 
-Supported for the PostgreSQL_ (``postgresql_psycopg2``) and MySQL_ (``mysql``)
-backends.
+Supported for the PostgreSQL_ (``postgresql``) and MySQL_ (``mysql``) backends.
```
Comment (charettes) [inline docs/ref/settings.txt]: While you're around I feel like `Supported by ...` makes more sense.

**Comment by jarshwah** [comment]
I agree that leaving the alias without a deprecation period just yet is a good idea. The overhead of maintaining it is basically nothing, and there is a wealth of information on the internet about configuring the pg backend with the old string that would all of a sudden be out of date. We can revisit in the future.

**Comment by caioariede** [comment]
Thanks guys! I updated the patch.

File: docs/releases/1.9.txt [hunk 5]
```diff
@@ -124,6 +124,12 @@ levels of graceful degradation.
 
 .. _YUI's A-grade: http://yuilibrary.com/yui/docs/tutorials/gbs/
 
+PostgreSQL backend renamed
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The PostgreSQL backend was renamed from ``django.db.backends.postgresql_psycopg2``
```
Comment (timgraham) [inline docs/releases/1.9.txt]: "was renamed from" -> "is also available from ... The old name will continue to be available for backwards compatibility."

File: docs/releases/1.9.txt [hunk 6]
```diff
@@ -124,6 +124,12 @@ levels of graceful degradation.
 
 .. _YUI's A-grade: http://yuilibrary.com/yui/docs/tutorials/gbs/
 
+PostgreSQL backend renamed
```
Comment (timgraham) [inline docs/releases/1.9.txt]: Please add a "Database backends" section to minor features and include this there.

File: docs/internals/deprecation.txt [hunk 7]
```diff
@@ -94,6 +94,9 @@ details on these changes.
 * The ``callable_obj`` keyword argument to
   ``SimpleTestCase.assertRaisesMessage()`` will be removed.
 
+* The ``django.db.backends.postgresql_psycopg2`` alias for the
```
Comment (timgraham) [inline docs/internals/deprecation.txt]: No more deprecation.

**Comment by caioariede** [comment]
Patch updated

**Comment by timgraham** [comment]
The backwards compatibility layer doesn't seem to work (see "default" build failure). Please also correct the flake8 and isort errors.

**Comment by caioariede** [comment]
I noticed `importlib.import_module` will fail against `django.db.backends.postgresql_psycopg2.base` when `base` is not a module, but a reference in the `postgresql_psycopg2.py` file. The `import_module` call is used in the [load_backend](https://github.com/django/django/blob/3e1bb5cfb8f9676e0c045f0d101a094437133b54/django/db/utils.py#L109) function.

I'm looking for an alternative solution. I think it might be better to use a function that accepts any kind of import paths, not only modules, to look for a fully qualified database backend name.

**Comment by timgraham** [comment]
I think I would add back the original files and import the moved files in them. This seems safer in case any third-party code imports the backend in a similar fashion.

**Comment by caioariede** [comment]
I thought about that too, but when doing something like:

**In:** `django/db/backends/postgresql/base.py`:

```python
from ..postgresql_psycopg2.base import *
```

Will make flake8 complain about using the wildcard. Importing everything explicitly does not look sane.

Not sure if specifying what should be imported from `base` is an option.

What do you think?

**Comment by timgraham** [comment]
You can use `# NOQA` on that line to silence the warning (as done elsewhere).

**Comment by caioariede** [comment]
@timgraham is it possible to force tests to run again?

It looks like to me some of them failed because some ephemeral issue. Not sure though.

**Comment by timgraham** [comment]
Yea, sometimes Jenkins has that hiccup.

Another idea: in `django.db.utils.load_backend()` add:
```
if backend_name == 'django.db.backends.postgresql_psycopg2':
    backend_name == 'django.db.backends.postgresql'
```
Then I think we could remove the postgresql_psycopg2 module (though maybe it makes sense to keep the modules for backwards compatibility for people importing them for their own pg backends -- we could do it in a separate commit so Git displays the change as file rename instead of massive additions and removals). That would allow us to put deprecation warnings in the shims without having them appear for most users simply using the string in their settings.

**Comment by caioariede** [comment]
The build looks good now

**Comment by timgraham** [comment]
merged in ec9004728ee136e3b7e2b7cd2610203e16b6ce9b and bcb4fe0012f8f869348ea83f5a35706f4545c44a, thanks!

**Pull request closed without merging**